### ISSUE TITLE
Deprecated options in cometbft config template

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/penumbra#3144]
**Describe the bug**

Recent smoke test runs showed
```
I[2023-10-03|21:23:02.579] deprecated usage found in configuration file usage="[fastsync] table detected. This section has been renamed to [blocksync]. The values in this deprecated section will be disregarded."
I[2023-10-03|21:23:02.579] deprecated usage found in configuration file usage="fast_sync key detected. This key has been renamed to block_sync. The value of this deprecated key will be disregarded."
```

This key is generated as part of `pd`'s config generation mechanism.

**Comment by conorsch** [comment]
Good call. I've generated a new default cometbft config via `cometbft init` and diffed against ours. The changes we want to pull in are pretty obvious, as are the exceptions.

**Comment by conorsch** [comment]
Looked into this some more. Turns out we can't perform the rename because `tendermint-config` hardcodes fastsync, so in order for us to leverage the strong typing of the config (#1827), we must have that key in config. I filed an upstream issue https://github.com/informalsystems/tendermint-rs/issues/1368. For now, I'll kick this into future queue so we can follow up later.
